### PR TITLE
JS: rename `getId()` to `getIdentifier()`

### PR DESCRIPTION
**Pull request by erik-krogh**
`getId()` and `getIdentifier()` was used inconsistently throughout our libraries. 

`getIdentifier()` was used more than `getId()`, and the result from most of the `getId()` predicate were `Identifiers`. 

I've renamed all `getId()` predicates that returned a `Identifier` to `getIdentifier()`.

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/Functions.qll [hunk 1]
```diff
@@ -78,7 +78,7 @@ class Function extends @function, Parameterized, TypeParameterized, StmtContaine
   }
 
   /** Gets the identifier specifying the name of this function, if any. */
-  VarDecl getId() { result = getChildExpr(-1) }
+  VarDecl getIdentifier() { result = getChildExpr(-1) }
```
Comment (esbena) [inline javascript/ql/src/semmle/javascript/Functions.qll:81]: This is a breaking change.

File: javascript/ql/src/semmle/javascript/TypeScript.qll [hunk 2]
```diff
@@ -10,11 +10,7 @@ class NamespaceDefinition extends Stmt, @namespacedefinition, AST::ValueNode {
   /**
    * Gets the identifier naming the namespace.
    */
-  Identifier getId() {
-    result = this.(NamespaceDeclaration).getId()
-    or
-    result = this.(EnumDeclaration).getIdentifier()
-  }
+  Identifier getIdentifier() { none() } // Overridden in subtypes.
```
Comment (esbena) [inline javascript/ql/src/semmle/javascript/TypeScript.qll:13]: This is a breaking change.

File: javascript/ql/src/semmle/javascript/TypeScript.qll [hunk 3]
```diff
@@ -179,12 +175,12 @@ class GlobalAugmentationDeclaration extends Stmt, StmtContainer, @globalaugmenta
 /** A TypeScript "import-equals" declaration. */
 class ImportEqualsDeclaration extends Stmt, @importequalsdeclaration {
   /** Gets the name under which the imported entity is imported. */
-  Identifier getId() { result = getChildExpr(0) }
+  Identifier getIdentifier() { result = getChildExpr(0) }
```
Comment (esbena) [inline javascript/ql/src/semmle/javascript/TypeScript.qll:178]: This is a breaking change.

File: javascript/ql/src/semmle/javascript/Functions.qll [hunk 4]
```diff
@@ -77,6 +77,13 @@ class Function extends @function, Parameterized, TypeParameterized, StmtContaine
     result = getDocumentation().getATagByTitle("this").getType()
   }
 
+  /**
+   * DEPRECATED: Use `getIdentifier()` instead.
+   *
+   * Gets the identifier specifying the name of this function, if any.
+   */
+  VarDecl getId() { result = getIdentifier() }
```
Comment (asgerf) [inline javascript/ql/src/semmle/javascript/Functions.qll:85]:
```suggestion
  deprecated
  VarDecl getId() { result = getIdentifier() }
```

File: javascript/ql/src/semmle/javascript/TypeScript.qll [hunk 5]
```diff
@@ -7,6 +7,13 @@ import javascript
  * considered to be namespace definitions.
  */
 class NamespaceDefinition extends Stmt, @namespacedefinition, AST::ValueNode {
+  /**
+   * DEPRECATED: Use `getIdentifier()` instead.
+   *
+   * Gets the identifier naming the namespace.
+   */
+  Identifier getId() { result = getIdentifier() }
```
Comment (asgerf) [inline javascript/ql/src/semmle/javascript/TypeScript.qll:15]:
```suggestion
  deprecated
  Identifier getId() { result = getIdentifier() }
```

File: javascript/ql/src/semmle/javascript/TypeScript.qll [hunk 6]
```diff
@@ -174,6 +181,13 @@ class GlobalAugmentationDeclaration extends Stmt, StmtContainer, @globalaugmenta
 
 /** A TypeScript "import-equals" declaration. */
 class ImportEqualsDeclaration extends Stmt, @importequalsdeclaration {
+  /**
+   * DEPRECATED: Use `getIdentifier()` instead.
+   *
+   * Gets the name under which the imported entity is imported.
+   */
+  Identifier getId() { result = getIdentifier() }
```
Comment (asgerf) [inline javascript/ql/src/semmle/javascript/TypeScript.qll:189]:
```suggestion
  deprecated
  Identifier getId() { result = getIdentifier() }
```

**Comment by erik-krogh** [comment]
@esbena can I get a final approve?

**Pull request merged**